Lib/asyncio/windows_events.py

```diff
@@ -308,6 +308,22 @@ def __init__(self, proactor=None):
             proactor = IocpProactor()
         super().__init__(proactor)
 
+    def run_forever(self):
+        try:
+            # if _self_reading_future is cancelled -
+            # this might indicate that event loop was interrupted before
+            # and self-reading routine is not hooked up now and needs
+            # to be restarted
+            if (self._self_reading_future is not None and
+                self._self_reading_future.cancelled()):
+                self._self_reading_future = None
+                self.call_soon(self._loop_self_reading)
+            super().run_forever()
+        except KeyboardInterrupt:
```

Comment:
Replace `except` with `finally`.
Canceling self-reading doesn't make harm if `run_forever()` was stopped by another exception (or just normal execution finishing).
We will re-schedule self-reading again on next `run_forever()` call.